src/redux/rockets/Rocket.js

```diff
@@ -0,0 +1,49 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const GET_DATA = 'GET_DATA';
+const RESERVE_ROCKET = 'RESERVE_ROCKET';
+const rocketsUrl = 'https://api.spacexdata.com/v4/rockets';
+
+const rocketsReducer = (state = [], action) => {
+  switch (action.type) {
+    case `${GET_DATA}/fulfilled`:
+      return action.payload;
+    case RESERVE_ROCKET:
+      return state.map((rocket) => {
+        if (rocket.id === action.payload) {
+          return {
+            ...rocket,
+            reserved: !rocket.reserved,
+          };
+        }
```

Comment:
- [ ] It is not stated explicitly in the project requirements, but please use two different actions to handle Booking/Canceling a rocket reservation, this is because having an "Action-based" structure, it doesn't make any sense that a user cancels a reservation firing a "Reserve" action. Also, this would give you better logs for later debugging! Please consider applying this. 🤓 🔧